src/main/java/dev/j3fftw/litexpansion/items/GlassCutter.java

```diff
@@ -0,0 +1,65 @@
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class GlassCutter extends SimpleSlimefunItem<ItemUseHandler> implements Listener {
+
+    private static final Material[] cuttableBlocks = { Material.GLASS, Material.WHITE_STAINED_GLASS, Material.ORANGE_STAINED_GLASS, Material.MAGENTA_STAINED_GLASS, Material.LIGHT_BLUE_STAINED_GLASS, Material.YELLOW_STAINED_GLASS, Material.LIME_STAINED_GLASS, Material.PINK_STAINED_GLASS, Material.GRAY_STAINED_GLASS, Material.LIGHT_GRAY_STAINED_GLASS, Material.CYAN_STAINED_GLASS, Material.PURPLE_STAINED_GLASS, Material.BLUE_STAINED_GLASS, Material.BROWN_STAINED_GLASS, Material.GREEN_STAINED_GLASS, Material.RED_STAINED_GLASS, Material.BLACK_STAINED_GLASS };
+
+
+    public GlassCutter() {
+        super(Items.LITEXPANSION, Items.GLASS_CUTTER, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            Items.REFINED_IRON, Items.REFINED_IRON, Items.REFINED_IRON,
+            new ItemStack(Material.SHEARS), Items.CARBON_PLATE, new ItemStack(Material.SHEARS),
+            null, Items.CARBON_PLATE, null
+        });
+
+        Bukkit.getPluginManager().registerEvents(this, LiteXpansion.getInstance());
+    }
+
+    public ItemUseHandler getItemHandler() {
+        return e -> e.setUseBlock(Event.Result.DENY);
+    }
+
+    @EventHandler
+    @SuppressWarnings("ConstantConditions")
+    public void onGlassCut(PlayerInteractEvent e) {
+        Player p = e.getPlayer();
+
+        if (e.getAction() == Action.LEFT_CLICK_BLOCK && SlimefunUtils.isItemSimilar(e.getItem(), Items.GLASS_CUTTER, true, false)) {
+            e.setCancelled(true);
+
+            SlimefunItem slimefunItem = BlockStorage.check(e.getClickedBlock());
+
+            if (slimefunItem != null) {
+                return;
+            }
+
+            for (Material cuttableBlock : cuttableBlocks) {
+                if (e.getClickedBlock().getType() == cuttableBlock) {
+                    ItemStack droppedGlass = new ItemStack(cuttableBlock);
+                    e.getClickedBlock().getLocation().getWorld().dropItemNaturally(e.getClickedBlock().getLocation(), droppedGlass);
+                    e.getClickedBlock().setType(Material.AIR);
```

Comment:
you don't even break out of this loop (which is redundant anyway) which means you'll waste CPU cydles as it'll still process even when it's found and executed the needed code.